### PR TITLE
first version of homework

### PR DESCRIPTION
**Pull request by KirillTimoshenko**

### CODE REVIEW

File: quadratic-equation/src/Main.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            String equationString = scanner.nextLine();
+            Pattern pattern = Pattern.compile(REGULAR_EXPRESSION);
+            Matcher matcher = pattern.matcher(equationString);
+            while (matcher.find()) {
+                coefficients.add(Double.parseDouble(matcher.group()));
+            }
+
+            if (coefficients.size() != 3) {
+                throw (new IllegalArgumentException("Not correct coefficients form."));
+            }
```
Comment (alex-prus) [inline quadratic-equation/src/Main.java:25]: Coefficients gathering logic better be in different method/class

File: quadratic-equation/src/QuadraticEquation.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        this.b = b;
+    }
+
+    public double getC() {
+        return c;
+    }
+
+    public void setC(double c) {
+        this.c = c;
+    }
```
Comment (alex-prus) [inline quadratic-equation/src/QuadraticEquation.java:37]: Why do you need getters/setters, they're unused and cannot be used in future as you don't need to change coefficients and get their values outside this class
Object of QuadraticEquation is final and represents the equation you need to solve
If you need to solve another equation, you create new object
So abc coefficients can be final

File: quadratic-equation/src/QuadraticEquation.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    public double getFirstRoot() {
+        return firstRoot;
+    }
+
+    public double getSecondRoot() {
+        return secondRoot;
+    }
+
+    public double getDiscriminant() {
+        return (Math.pow(b, 2) - 4 * a * c);
```
Comment (alex-prus) [inline quadratic-equation/src/QuadraticEquation.java:48]: excess ()

File: quadratic-equation/src/QuadraticEquation.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    public void solve() throws NegativeDiscriminantException {
+        double discriminant = getDiscriminant();
+        if (discriminant > 0.0) {
+            firstRoot = (-b + Math.sqrt(discriminant)) / (2 * a);
+            secondRoot = (-b - Math.sqrt(discriminant)) / (2 * a);
+        } else if (discriminant == 0.0) {
+            firstRoot = -b / (2 * a);
+            secondRoot = firstRoot;
+        } else {
+            throw (new NegativeDiscriminantException("Negative discriminant." +
```
Comment (alex-prus) [inline quadratic-equation/src/QuadraticEquation.java:60]: excess ()

File: quadratic-equation/src/Main.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+            System.out.println("Write the coefficients of the equation in following form: (a, b, c)");
+            String equationString = scanner.nextLine();
+            Pattern pattern = Pattern.compile(REGULAR_EXPRESSION);
+            Matcher matcher = pattern.matcher(equationString);
+            while (matcher.find()) {
+                coefficients.add(Double.parseDouble(matcher.group()));
+            }
+
+            if (coefficients.size() != 3) {
+                throw (new IllegalArgumentException("Not correct coefficients form."));
```
Comment (alex-prus) [inline quadratic-equation/src/Main.java:24]: excess ()

File: quadratic-equation/src/Main.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Main {
+
+    private static final String REGULAR_EXPRESSION = "(-?\\d+(\\.\\d+)?)";
```
Comment (alex-prus) [inline quadratic-equation/src/Main.java:9]: Hope you fully understand this expression

File: quadratic-equation/src/Main.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+    public static void main(String[] arg) {
+        try {
+            Scanner scanner = new Scanner(System.in);
+            List<Double> coefficients = new ArrayList<>();
+            System.out.println("Write the coefficients of the equation in following form: (a, b, c)");
+            String equationString = scanner.nextLine();
+            Pattern pattern = Pattern.compile(REGULAR_EXPRESSION);
```
Comment (alex-prus) [inline quadratic-equation/src/Main.java:17]: I don't see the point to use here regular expression, more readable would be with simple 3 separate inputs
Regular expression might be useful if you parse the whole equation string like (ax2+bx+c=0)

**Comment by KirillTimoshenko** [comment]
Added some commits.  Waiting for feedback.

File: quadratic-equation/src/QuadraticEquation.java [hunk 8]
```diff
@@ -44,21 +20,21 @@ public double getSecondRoot() {
         return secondRoot;
     }
 
-    public double getDiscriminant() {
-        return (Math.pow(b, 2) - 4 * a * c);
-    }
-
     public void solve() throws NegativeDiscriminantException {
-        double discriminant = getDiscriminant();
+        double discriminant = countDiscriminant();
```
Comment (Neaved) [inline quadratic-equation/src/QuadraticEquation.java:24]: count? Count means counting: first Discriminant, Second Discriminant etc.
here u need calculate Discriminant value

File: quadratic-equation/src/QuadraticEquation.java [hunk 9]
```diff
@@ -1,39 +1,15 @@
 public class QuadraticEquation {
 
-    private double a;
-    private double b;
-    private double c;
+    private final double A;
+    private final double B;
+    private final double C;
```
Comment (Neaved) [inline quadratic-equation/src/QuadraticEquation.java:5]: why did they become capital letters?

File: quadratic-equation/src/CoefficientsScanner.java [hunk 10]
```diff
@@ -0,0 +1,21 @@
+import java.util.ArrayList;
+import java.util.InputMismatchException;
+import java.util.List;
+import java.util.Scanner;
+
+public class CoefficientsScanner {
+
+    public QuadraticEquation scanEquationCoefficients() throws InputMismatchException {
```
Comment (Neaved) [inline quadratic-equation/src/CoefficientsScanner.java:8]: method name do not correspond return value

here u enter coefficients, so then just return them back

**Comment by KirillTimoshenko** [comment]
commit changes. waiting for feedback

**Pull request merged**